src/cluster/Cluster.cc

```diff
@@ -298,3 +427,72 @@ int Cluster::from_xml(const string& xml)
 
     return 0;
 }
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+void Cluster::set_error_message(const char *mod, const string& name, const string& message)
+{
+    static const int MAX_ERROR_MSG_LENGTH = 100;
+
+    SingleAttribute * attr;
+    ostringstream     error_value;
+
+    error_value << one_util::log_time() << ": " << message.substr(0, MAX_ERROR_MSG_LENGTH);
+
+    if (message.length() >= MAX_ERROR_MSG_LENGTH)
+    {
+        error_value << "... see more details in oned.log";
+
+        NebulaLog::error(mod, message);
+    }
+
+    attr = new SingleAttribute(name, error_value.str());
+
+    obj_template->erase(name);
+    obj_template->set(attr);
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+void Cluster::load_plan()
+{
+    auto plpool = Nebula::instance().get_planpool();
+
+    if (auto plan = plpool->get_ro(oid))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that you are using unintended operations or assignments that could lead to logical errors, vulnerabilities, or undefined behavior. To prevent these issues, ensure that you follow best practices, such as using explicit comparisons and clear operator usage, and using parentheses to clarify intent. Learn more - https://cvefeed.io/cwe/detail/cwe-480-use-of-incorrect-operator